test/NumberGeneratorTrait.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+declare(strict_types=1);
+
+namespace LaminasTest\I18n\PhoneNumber;
+
+use Generator;
+use Laminas\I18n\PhoneNumber\PhoneNumberValue;
+use libphonenumber\PhoneNumberType;
+use libphonenumber\PhoneNumberUtil;
+use libphonenumber\ShortNumberInfo;
+
+use function assert;
+use function is_int;
+use function is_string;
+use function sprintf;
+
+/**
+ * @internal \LaminasTest\I18n
+ */
+trait NumberGeneratorTrait
```

Comment:
`libphonenumber`'s apparently valid example numbers don't all validate. These data providers work-around some of the unexpected numbers that get returned - typically, we're filtering out tens of results from many hundreds-ish.